File: Python/Introductory/IncreasingArray.py
#!/usr/bin/env python
# coding: utf-8

# In[10]:


array_size = input()
array = list(map(int,input().split()))
def increasing_array(array, array_size):
    prev_num = -1
    total_moves = 0
    for num in array:        
        if num <= prev_num:
            moves = prev_num - num
            total_moves += moves
            prev_num = num + moves
        else:
            prev_num = num
    return total_moves
print(increasing_array(array, array_size))

File: C++/Introductory/IncreasingArray.cpp
#include <iostream>
using namespace std;

int main()
{
    int size;
    cin >> size;
    long long moves = 0;
    long long curr_num;
    long long prev_num;
    cin >> prev_num;

    for (int i = 1; i < size; i++)
    {
        cin >> curr_num;
        if (curr_num < prev_num)
            moves += prev_num - curr_num;
        prev_num = max(curr_num, prev_num);
    }

    cout << moves << endl;
}